python/tests/test_deserialization.py

```diff
@@ -0,0 +1,279 @@
+from typing import Any
+
+import pytest
+from scylla._rust.session_builder import SessionBuilder  # pyright: ignore[reportMissingModuleSource]
+
+from scylla._rust.results import RowFactory, ColumnIterator  # pyright: ignore[reportMissingModuleSource]
+
+
+@pytest.mark.asyncio
+async def test_simple_deserialization():
+    # 1. Connect
+    builder = SessionBuilder(["127.0.0.2"], 9042)
+    session = await builder.connect()
+
+    # 2. Create keyspace & table
+    await session.execute("""
+        CREATE KEYSPACE IF NOT EXISTS testks
+        WITH replication = {'class': 'SimpleStrategy', 'replication_factor': 1};
+    """)
+
+    await session.execute("""
+                          CREATE TABLE IF NOT EXISTS testks.example_table (
+                                                                              id int PRIMARY KEY,
+                                                                              value_int int,
+                                                                              value_text text
+                          );
+                          """)
+
+    # 3. Insert test rows
+    await session.execute("""
+                          INSERT INTO testks.example_table (id, value_int, value_text)
+                          VALUES (1, 42, 'hello');
+                          """)
+
+    await session.execute("""
+                          INSERT INTO testks.example_table (id, value_int, value_text)
+                          VALUES (2, 99, 'world');
+                          """)
+
+    # 4. Query the data
+    result = await session.execute("SELECT * FROM testks.example_table")
+
+    # 5. Print CQL values (should not throw)
+    rows = result.create_rows_result()
+    for row in rows:
+        print(row)
+
+
+@pytest.mark.asyncio
+async def test_list_deserialization():
+    # 1. Connect
+    builder = SessionBuilder(["127.0.0.2"], 9042)
+    session = await builder.connect()
+
+    # 3. Create a table with complex CQL types
+    await session.execute("""
+                          CREATE TABLE IF NOT EXISTS testks.complex_table2 (
+                                                                              id int PRIMARY KEY,
+                                                                              name text,
+                                                                              scores list<int>,
+                          );
+                          """)
```

Comment:
:wrench: Can we make this test use nested lists? Just to have this additional case tested.